include/qunit.hpp

```diff
@@ -56,7 +56,10 @@ class QUnit : public QInterface {
     virtual void SetQuantumState(complex* inputState);
     virtual void GetQuantumState(complex* outputState);
     virtual complex GetAmplitude(bitCapInt perm);
-    virtual void SetPermutation(bitCapInt perm) { SetReg(0, qubitCount, perm); }
+    virtual void SetPermutation(bitCapInt perm, complex phaseFac = complex(-999.0, -999.0))
```

Comment:
This parameter is necessary for the prototype that the QEngine and QFusion types use as a convenience. It's harder to respect this parameter for QUnit, because each shard has its own phase factor, and a global phase factor is the product of all shard phase factors, which is not easily set.

This kind of initial phase factor is pseudo-quantum. It has no physical meaning, in effect. It's a convenience for simulation. If we absolutely need to respect to it in all cases, I would sooner remove it entirely, instead.

However, other projects we might interface with (for example, namely ProjectQ) don't treat initial global phase as physically meaningless, as we have been careful to do. So, we're caught between theoretical physics best practices and fully supporting the expectations of other quantum computer simulator software. Please advise.